src/shared/extensionUtilities.ts

```diff
@@ -69,3 +69,23 @@ export function safeGet<O, T>(obj: O | undefined, getFn: (x: O) => T): T | undef
 
     return undefined
 }
+
+/**
+ * Utility function to search for tokens in a string and convert them to relative paths parseable by VS Code
+ * Useful for converting HTML images to webview-usable images
+ *
+ * @param extPath Extension path (from extension context)
+ * @param fileText Text of the file to scan
+ * @param token Token (string or regex) to search for (default: `/!!EXTENSIONROOT!!/g`)
+ */
+export function convertPathTokensToPath(
+    extPath: string,
+    fileText: string,
+    token: RegExp | string = /!!EXTENSIONROOT!!/g
```

Comment:
Nit: If strings will just be converted (unescaped) to regexes anyway, suggest to remove `string` from the type union.

Also, minor naming suggestions:

```typescript
export function resolveResourcePaths(
    basePath: string,
    markdown: string,
    pattern: Regex = /!!EXTENSIONROOT!!/g
): string { ... }
```

Reply:
`markdown` is inaccurate because it's already been converted to HTML at this point. Will switch to `text` since this can be applied to other things as well.